### PR TITLE
otlpmetricgrpc: fix preference for user supplied dialOptions

### PR DESCRIPTION
**Pull request by ps-mir**
## Overview
User supplied grpc.DialOptions when supplied with otlpmetricgrpc.WithDialOption is being dropped because `NewGRPCConfig` applies its own internal defaults after applying user options.

This led to internally computed entries land (and override) after user-supplied ones, leading to drop of some settings like transport credentials.

## Fix
This is the same bug and fix shape as #8805 (otlptracegrpc). One structural difference: otlpmetricgrpc never had `grpc.WithUserAgent` inside oconf's DialOptions to begin with (it's applied separately in client.go), so dialOptsPrefix here only needs to hold the service-config/credentials/compression/reconnect defaults.

Fixes
* #8822 

Related to
* #8805

### CODE REVIEW

File: internal/shared/otlp/otlpmetric/oconf/options.go.tmpl [hunk 1]
```diff
@@ -144,30 +144,36 @@ func NewGRPCConfig(opts ...GRPCOption) Config {
...
-		cfg.DialOptions = append(cfg.DialOptions, grpc.WithDefaultCallOptions(grpc.UseCompressor(gzip.Name)))
+		dialOptsPrefix = append(dialOptsPrefix, grpc.WithDefaultCallOptions(grpc.UseCompressor(gzip.Name)))
 	}
 	if cfg.ReconnectionPeriod != 0 {
 		p := grpc.ConnectParams{
 			Backoff:           backoff.DefaultConfig,
 			MinConnectTimeout: cfg.ReconnectionPeriod,
 		}
-		cfg.DialOptions = append(cfg.DialOptions, grpc.WithConnectParams(p))
+		dialOptsPrefix = append(dialOptsPrefix, grpc.WithConnectParams(p))
```
Comment (pellared) [inline internal/shared/otlp/otlpmetric/oconf/options.go.tmpl:174]: Could we add a shared-template unit test covering non-empty `ServiceConfig` and non-zero `ReconnectionPeriod`? This PR moves both branches into `dialOptsPrefix`, but the current test exercises only credentials, leaving the generated gRPC and HTTP copies uncovered and causing `codecov/patch` to fail at 75%. A focused `NewGRPCConfig` test verifying these defaults remain included alongside a caller-supplied DialOption should be sufficient; a timing-based reconnection integration test is unnecessary.

**Review by pellared — COMMENTED** [review]
One test coverage suggestion. Otherwise, LGTM

File: internal/shared/otlp/otlpmetric/oconf/options.go.tmpl (hunk 1 above)
Comment (ps-mir) [reply internal/shared/otlp/otlpmetric/oconf/options.go.tmpl:174]: [4bfc67c](https://github.com/open-telemetry/opentelemetry-go/pull/8834/commits/4bfc67c888c7d3211bb2b23561d627ea2e4664f2)

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by dashpole — APPROVED** [review]
